pkg/reconciler/trigger/trigger.go

```diff
@@ -61,43 +68,51 @@ type TriggerReconciler struct {
 	uriResolver        *resolver.URIResolver
 
 	// projectID is used as the GCP project ID when present, skipping the
-	// metadata server check. Used by tests.
+	// metadata server check.
 	projectID string
 
 	// pubsubClient is used as the Pubsub client when present.
 	pubsubClient *pubsub.Client
 }
 
 // Check that TriggerReconciler implements Interface
-var _ triggerreconciler.Interface = (*TriggerReconciler)(nil)
-var _ triggerreconciler.Finalizer = (*TriggerReconciler)(nil)
-
-func (r *TriggerReconciler) ReconcileKind(ctx context.Context, t *brokerv1beta1.Trigger) pkgreconciler.Event {
-	b := brokerFromContext(ctx)
-	if b == nil {
-		// Assume the Broker has been deleted or doesn't exist yet. Create a
-		// Broker object with the expected name and namespace to
-		// reconcile with.
-		// TODO move this to resources package?
-		b = &brokerv1beta1.Broker{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      t.Spec.Broker,
-				Namespace: t.Namespace,
-			},
-			//TODO is this needed?
-			Status: brokerv1beta1.BrokerStatus{},
-		}
-		b.Status.InitializeConditions()
-		//return fmt.Errorf("Couldn't fetch Broker from context")
+var _ triggerreconciler.Interface = (*Reconciler)(nil)
+var _ triggerreconciler.Finalizer = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (r *Reconciler) ReconcileKind(ctx context.Context, t *brokerv1beta1.Trigger) pkgreconciler.Event {
+	b, err := r.brokerLister.Brokers(t.Namespace).Get(t.Spec.Broker)
+
+	// If there is an error getting the Broker or the Broker is deleted, we mark the Trigger status
+	// regardless of whether the Trigger belongs to this controller or not. It doesn't hurt.
```

Comment:
If multiple trigger controllers are doing this and they disagree on the status message, then they'll fight over the message forever. This should probably check for the condition name and status (BrokerReady, Unknown) before attempting to set. Ultimately we'll want a shared trigger controller in eventing doing this.